src/contour/Config.h

```diff
@@ -144,263 +189,3020 @@ struct SimpleColorConfig
+        "# whether or not to put the terminal's screen into fullscreen mode.\n"
+        "# It is activated during startup as well as when switching from another profile to this one.\n"
+        "fullscreen: {}\n"
+        "\n"
+    };
+    ConfigEntry<bool> showTitleBar { true,
+                                     "# When this profile is *activated*, this flag decides\n"
+                                     "# whether or not the title bar will be shown\n"
+                                     "show_title_bar: {}\n"
+                                     "\n" };
+    ConfigEntry<bool> sizeIndicatorOnResize { true, "\n" };
+    ConfigEntry<bool> mouseHideWhileTyping { true,
+                                             "# whether or not to hide mouse when typing\n"
+                                             "hide_while_typing: {}\n"
+                                             "\n" };
+    ConfigEntry<vtbackend::RefreshRate> refreshRate { 0.0, "fmt formatted doc {} \n" };
+    ConfigEntry<vtbackend::LineOffset> copyLastMarkRangeOffset = { 0, "fmt formatted doc {} \n" };
+    ConfigEntry<string> wmClass = { "contour",
+                                    "# Defines the class part of the WM_CLASS property of the window.\n" };
+    ConfigEntry<vtbackend::PageSize> terminalSize { { vtbackend::LineCount(25), vtbackend::ColumnCount(80) },
+                                                    "#Determines the initial terminal size in  characters\n"
+                                                    "terminal_size:\n"
+                                                    "    columns: {} \n"
+                                                    "    lines: {} \n"
+                                                    "\n" };
+    ConfigEntry<vtbackend::VTType> terminalId { vtbackend::VTType::VT525,
+                                                "# Determines the terminal type that is being advertised.\n"
+                                                "# Possible values are:\n"
+                                                "#   - VT100\n"
+                                                "#   - VT220\n"
+                                                "#   - VT240\n"
+                                                "#   - VT330\n"
+                                                "#   - VT340\n"
+                                                "#   - VT320\n"
+                                                "#   - VT420\n"
+                                                "#   - VT510\n"
+                                                "#   - VT520\n"
+                                                "#   - VT525\n"
+                                                "terminal_id: {}\n"
+                                                "\n" };
+    ConfigEntry<vtbackend::MaxHistoryLineCount> maxHistoryLineCount {
+        vtbackend::LineCount(1000),
+        "# Number of lines to preserve (-1 for infinite).\n"
```

Comment:
can we get the syntax of a potential serialization format (like YAML) out of the documentation?

The config file writer, that is about to write the comments above each key, should make sure it's proper syntax in the target format, like YAML. so a comment like `"this makes foo"` should become `"    # this makes foo\n"` with leading spaces to do the indentation proper, and `#` to introduce the YAML comment. the last line of a multi line comment should not require to have an LF. that's easy to add.

This makes it free for us to exchange YAML in the future with something else, say, ... LUA? or TOML etc :)

Reply:
Hm, TOML has the same comment character, LUA does not, maybe we can add this generalization in the future, what do you think?

Reply:
but we should not apply this is to every line, for example here:
```
"# whether or not to hide the scrollbar when in alt-screen.\n"
"hide_in_alt_screen: {}\n"
 ```
 we need to add comment only in first line, for that we need to have something like 
```
"{comment} whether or not to hide the scrollbar when in alt-screen.\n"
"hide_in_alt_screen: {}\n"
 ```
 and then format it with `fmt::arg("comment", "#")` maybe its not that difficult so i can add it then

Reply:
I think defaults should not be mentioned in C++'s documentation at all. In fact, that could be inferred and auto-generated in the docs, IMHO. No need to encode that into the documentation field, if it's already expressed in the C++ struct. That should work, shouldn't it?

Reply:
> I think defaults should not be mentioned in C++'s documentation at all. In fact, that could be inferred and auto-generated in the docs, IMHO. No need to encode that into the documentation field, if it's already expressed in the C++ struct. That should work, shouldn't it?

You are right, better not place default values there explicitly, will fix it, default construction of config gives exactly the default config that we have in `contour.yml` file